sycl/doc/extensions/supported/sycl_ext_intel_dataflow_pipes.asciidoc

```diff
@@ -170,33 +186,43 @@ The pipe class exposes static member functions for writing a data word to a pipe
 
 Blocking and non-blocking forms of the read and write members are defined, with the form chosen based on overload resolution.
 
+In the future, the `sycl::memory_order` parameter of read/write functions will control how other memory accesses, including regular, non-atomic memory accesses, are to be ordered around the pipe read/write operation.  The default memory order is `sycl::memory_order::seq_cst`. Currently, `sycl::memory_order` parameter is defined but not being used in the implementation.
+
 [source,c++,Read write members,linenums]
 ----
-template <typename Name,
-          typename DataT,
-          size_t MinCapacity = 0>
+template <class _name, class _dataT, int32_t _min_capacity = 0,
+          class _propertiesT = decltype(oneapi::experimental::properties{}),
+          class = void>
 class pipe {
   // Blocking
-  static DataT read();
-  static void write( const DataT &Data );
+  static _dataT read();
+  static void write(const _dataT &Data) ;
 
   // Non-blocking
-  static DataT read( bool &Success );
-  static void write( const DataT &Data, bool &Success );
+  static _dataT read(bool &Success);
+  static void write(const _dataT &Data, bool &Success) ;
 
   // Static members
-  using value_type = DataT;
-  size_t min_capacity = MinCapacity;
+  static constexpr int32_t m_Size = sizeof(_dataT);
+  static constexpr int32_t m_Alignment = alignof(_dataT);
+  static constexpr int32_t m_Capacity = _min_capacity;
+  static constexpr int32_t m_ready_latency = oneapi::experimental::detail::ValueOrDefault<_propertiesT, ready_latency_key>::template get<int32_t>(0);
+  static constexpr int32_t m_bits_per_symbol = oneapi::experimental::detail::ValueOrDefault<_propertiesT, bits_per_symbol_key>::template get<int32_t>(8);
+  static constexpr bool m_uses_valid = oneapi::experimental::detail::ValueOrDefault<_propertiesT, uses_valid_key>::template get<bool>(true);
+  static constexpr bool m_first_symbol_in_high_order_bits = oneapi::experimental::detail::ValueOrDefault<_propertiesT, first_symbol_in_high_order_bits_key>::template             get<int32_t>(0);
+  static constexpr protocol_name m_protocol = oneapi::experimental::detail:: ValueOrDefault<_propertiesT, protocol_key>::template get<protocol_name>(
+          protocol_name::AVALON_STREAMING_USES_READY);
```

Comment:
Are these meant to be public interfaces? If so, they should follow typical SYCL member naming.

Additionally, if they are indeed supposed to be public, please void using unspecified (and in this case `detail`) functionality. Instead you can use

```c++
... = __unspecified__;
```

and either specify the intended behavior in a comment or in the description below.

Reply:
Thanks @steffenlarsen for your review. What's the definition of the public interface?

The code lives in `sycl/include/sycl/ext/intel/experimental/pipes.hpp` , but those variables are internally used. The public interface which customer will use is similar though.

For example: they can use this to define `m_uses_valid` is true.

```
using default_pipe_properties = decltype(sycl::ext::oneapi::experimental::properties(sycl::ext::intel::experimental::uses_valid<true>));
```

Similar to the previous comments, the following code was the existing code in the `pipes.hpp`.

```
  static constexpr int32_t m_Size = sizeof(_dataT);
  static constexpr int32_t m_Alignment = alignof(_dataT);
  static constexpr int32_t m_Capacity = _min_capacity;
```

I made the change to the Spec was due to we moved the code from `hostpipes.hpp` to the existing `pipes.hpp`, and changed the Spec to match the existing unchanged code in the `pipes.hpp`

Reply:
> Thanks @steffenlarsen for your review. What's the definition of the public interface?

The public interface is anything the users may access. Anything not part of the public interface should not be included in the extension (can be included in the design document if needed).

> The code lives in `sycl/include/sycl/ext/intel/experimental/pipes.hpp` , but those variables are internally used. The public interface which customer will use is similar though.
> 
> For example: they can use this to define `m_uses_valid` is true.
> 
> ```
> using default_pipe_properties = decltype(sycl::ext::oneapi::experimental::properties(sycl::ext::intel::experimental::uses_valid<true>));
> ```

As long as the one the users have access to is named `uses_valid` and not `m_uses_valid`, it is in line with other SYCL interfaces. Having the `m_` prefix is not.

> Similar to the previous comments, the following code was the existing code in the `pipes.hpp`.
> 
> ```
>   static constexpr int32_t m_Size = sizeof(_dataT);
>   static constexpr int32_t m_Alignment = alignof(_dataT);
>   static constexpr int32_t m_Capacity = _min_capacity;
> ```
> 
> I made the change to the Spec was due to we moved the code from `hostpipes.hpp` to the existing `pipes.hpp`, and changed the Spec to match the existing unchanged code in the `pipes.hpp`

In the current variant of sycl/include/sycl/ext/intel/experimental/pipes.hpp these are private and as such are not user-visible. If that is changing, we need to rename these.

Reply:
@steffenlarsen , same question as above, do you mean I could change the naming here in the Spec but leave the variable name unchanged in the `pipes.hpp`?

Reply:
@steffenlarsen , thanks for clarifying the public interface means the `public` in C++ concept. All of those variables are private static variables, they will stay private, so I believe we don't need to rename anything here or do any change here.

Reply:
The extension should not mention anything that isn't visible to the user, so they should be removed from here if they are private.